packages/next/src/utils.ts

```diff
@@ -1254,21 +1254,38 @@ async function getSourceFilePathFromPage({
     const extensionless = fsPath.replace(path.extname(fsPath), '');
 
     for (const ext of extensionsToTry) {
-      fsPath = `${extensionless}.${ext}`;
+      // for appDir, we need to treat "index.js" as root-level "page.js"
```

Comment:
Should this special case be wrapped in `if (pageType === 'app')`?

Reply:
applied the suggestion 👍